### ISSUE TITLE
Why is SetRows doubled?

### ISSUE THREAD

**Issue by RVXD** [Firesphere/silverstripe-solr-search#134]
Sorry, not sure if this is a bug... But at least it's bugging me...

**Using setRows on BaseQuery, causes Results to get doubled.**

**Code that causes this:**
QueryComponentFactory.php
line: 92

```
// Double the rows in case something has been deleted, but not from Solr
$this->clientQuery->setRows($this->query->getRows() * 2);
```

**Question**
Why are the results doubled. Wouldn't it be easier to clean up the Solr in the indexing task?

**PaginatedMatches**
PaginatedMatches is not working correctly, also doubles the result. Can be fixed in template by adding the same .Limit(9) again. Not sure if that was the intended way of handling things.

```
$query = new BaseQuery();
$query->setStart($start);
$query->setRows(9); // causes 18 results
```

```
<ul>
   <% loop $Results.Limit(9) %>
      <li>
         <% include SilverShop/Includes/ProductGroupItem %>
      </li>
   <% end_loop %>
 </ul>
<% with $Results %>
    <% include Pagination %>
<% end_with %>
```
**Thank you for this nice module!**

**Comment by Firesphere** [comment]
Hi @RXVD,

The rows are doubled, to avoid possible errors around canView being false. This is a left-over from the module called "FulltextSearch", which filters canView after Solr returns it's results.
This module attempts to do this filtering on a Solr level, reducing the need for this check on the PHP side. However, for the sake of being on the safe side, this module returns double the amount of results, so it can be filtered if needed.

The bug you're encountering however, is a proper bug. It's not in the amount of results returned, but in the ResultSet returned. Thank you for raising this! I've added this to the project.

**Comment by Firesphere** [comment]
p.s.
Apologies for the delayed response. We've been quite busy and can't reply to every Issue in time, sadly.

**Comment by Firesphere** [comment]
@RVXD 
Could you please supply the relevant parts of your controller that's causing the problem?

It would help debug the issue greatly :)

**Comment by RVXD** [comment]
I use this in my page controller:

```
public function search(HTTPRequest $request)
    {
        $searchTerm = $request->getVar('search');

        // search products ************************************
        $start = 0;
        $paginationVar = 'start_products';
        if( !empty($_GET[$paginationVar]) ){
            $start = (int) $_GET[$paginationVar];
        }

        $query = new BaseQuery();
        $query->setStart($start);
        $query->setRows(12);
        $query->setSpellcheck(true);
        $query->addTerm($searchTerm);
        $query->addFilter('ClassName',Product::class);
        $query->addFilter('SiteTree_AllowPurchase',1);
        $query->addExclude('SiteTree_ImageID',0);

        $baseIndex = new SiteTreeIndex();
        $result = $baseIndex->doSearch($query);
        if( $result ){
            $products = $result->getPaginatedMatches($request);
            $products->setPaginationGetVar($paginationVar);
        } else {
            $products = new ArrayList();
        }
}
```
In my template I use this, with limit(12) added, it omitted, things go wrong:

```
                <h2>U heeft gezocht naar '$SearchTerm'</h2>
                <% if $Products.TotalItems %>
                    <h3>Gevonden artikelen ($Products.TotalItems)</h3>
                    <div class="grid-x grid-padding-x small-up-2 medium-up-3 large-up-4">
                        <% loop $Products.Limit(12) %>
                            <div class="cell">
                                <% include SilverShop/Includes/ProductGroupItem %>
                            </div>
                        <% end_loop %>
                    </div>
                    <% with $Products %>
                        <% include Pagination %>
                    <% end_with %>
                <% end_if %>
```


Hope it is of any use.

**Comment by Firesphere** [comment]
This is due to some weird logic in PaginatedList, it seems. And the wonderfully undocumented method `setLimitItems()`.

In the PaginatedResults, this is set to `false`, because we paginate in Solr. Setting it to `true` will cause the PaginatedList to attempt to grab the subset of items itself (causing skipped items, e.g. page 2 will display those from page 3, page 3 will display nothing anymore)

Your workaround is correct, for now, I'm fixing it and it will probably be in the next minor release

**Comment by Firesphere** [comment]
Closing this issue as it is fixed in #137 , future discussion should happen there.

Cheers @RVXD !

**Comment by Firesphere** [comment]
Update to #139 due to unexpected test breaking
